### PR TITLE
mimir-1291 feil i engelsk datoformat publikasjonsarkivet

### PR DESCRIPTION
**Pull request by Carl-OW**
"Enhanced the calculateQuarter function to correctly display suffixes for the English version."

### CODE REVIEW

File: src/main/resources/lib/ssb/utils/variantUtils.ts [hunk 1]
```diff
@@ -139,13 +139,32 @@ function calculateQuarter(previousFrom: string, language: string): string {
   const date: Date = new Date(previousFrom)
   const fromMonth: number = date.getMonth() + 1
   const quarter: number = Math.ceil(fromMonth / 3)
+
+  if (language === 'en') {
+    const suffix = getSuffix(quarter)
+    return `figures for ${quarter}${suffix} Quarter ${date.getFullYear()}`
```
Comment (johnnadeluy) [inline src/main/resources/lib/ssb/utils/variantUtils.ts:145]: This string will overwrite the translation in the english phrases. Meaning that if this phrase get changed in the phrases file, the change won't be reflected. Se suggestion below.

File: src/main/resources/lib/ssb/utils/variantUtils.ts [hunk 2]
```diff
@@ -139,13 +139,32 @@ function calculateQuarter(previousFrom: string, language: string): string {
...
+  }
+
   return localize({
     key: 'period.quarter',
     locale: language,
     values: [quarter.toString(), date.getFullYear().toString()],
```
Comment (johnnadeluy) [inline src/main/resources/lib/ssb/utils/variantUtils.ts:151]: You should instead replace the quarter.toString() string in values in localize when the language is english.

Explanation: 
This is how period.quarter looks like in the phrases_en.properties file = figures for {0}. quarter {1}. The first string in the values array will replace {0}, whilst the second string will replace {1} etc.

**Review by johnnadeluy — CHANGES_REQUESTED** [review]
See the comments above.

File: src/main/resources/lib/ssb/utils/variantUtils.ts (hunk 1 above)
Comment (Carl-OW) [reply src/main/resources/lib/ssb/utils/variantUtils.ts]: added changes so phrases_en.properties does not get overwritten

File: src/main/resources/lib/ssb/utils/variantUtils.ts (hunk 2 above)
Comment (Carl-OW) [reply src/main/resources/lib/ssb/utils/variantUtils.ts:155]: removed the . after {0}, and added localization for en version

File: src/main/resources/lib/ssb/utils/variantUtils.ts [hunk 3]
```diff
@@ -139,13 +139,36 @@ function calculateQuarter(previousFrom: string, language: string): string {
...
   })
 }
 
+function getSuffix(quarter: number): string {
+  switch (quarter) {
+    case 1:
```
Comment (Glenruben) [inline src/main/resources/lib/ssb/utils/variantUtils.ts:161]: We need to also correctly show ordinal numbers larger than 3. 
```case 1: 
case 21:
case 31:
  return 'st'
```
and so on

Comment (Glenruben) [reply src/main/resources/lib/ssb/utils/variantUtils.ts:161]: oh no this is for quarters not dates. My bad. Never mind.

**Review by Glenruben — APPROVED** [review]
Neat and tidy :)

File: src/main/resources/lib/ssb/parts/publicationArchive.ts [hunk 4]
```diff
@@ -133,7 +133,7 @@ function prepareStatisticRelease(
 
     return {
       title: language === 'en' ? release.statisticNameEn : release.statisticName,
-      period: period.charAt(0).toUpperCase() + period.slice(1),
+      period: period.charAt(0) + period.slice(1),
```
Comment (johnnadeluy) [inline src/main/resources/lib/ssb/parts/publicationArchive.ts:136]: It's enough to define `period` here. The rest of the code removes the first letter of the string and re-adds it, which is not necessary.

**Pull request merged**